### PR TITLE
AM2R: fix Lower Factory Intersection climb

### PR DESCRIPTION
**Pull request by JeffGainsNGames**
- add dboost for basic walljump
- add short charge spark method

### CODE REVIEW

**Comment by JeffGainsNGames** [comment]
@Miepee do we have a standard difficulty rating for short charges? (noting if this is advanced it's the first use)

File: randovania/games/am2r/logic_database/Industrial Complex.txt [hunk 1]
```diff
@@ -1108,7 +1108,19 @@ Extra - minimap_data: [{'x': 67, 'y': 29}, {'x': 67, 'y': 30}, {'x': 68, 'y': 28
   * Layers: default
   * 4-Tiles High Dock to Lower Factory Lava Lake Access/Horizontal Dock to Lower Factory Intersection
   > Horizontal Dock to Lower Factory Hub
-      Hijump With Power Grip Wall
+      Any of the following:
+          Space Jump or Can Use Spider Ball
```
Comment (Miepee) [inline randovania/games/am2r/logic_database/Industrial Complex.txt:1112]: spider ball needs no room rando or mid air morph/bombs/spring or something like that.

File: randovania/games/am2r/logic_database/Industrial Complex.txt [hunk 2]
```diff
@@ -1109,18 +1109,21 @@ Extra - minimap_data: [{'x': 67, 'y': 29}, {'x': 67, 'y': 30}, {'x': 68, 'y': 28
...
           All of the following:
               Hi-Jump Boots
               Power Grip or Walljump (Beginner)
           # IBJ
           Infinite Bomb Jumping (Intermediate) and Can Use Bombs
           # CBJ
           Charged Bomb Jump (Intermediate) and Can Use Charged Bomb Jump
+          All of the following:
+              Can Use Spider Ball
+              Bombs or Spring Ball or Mid-Air Morph (Beginner)
```
Comment (Miepee) [inline randovania/games/am2r/logic_database/Industrial Complex.txt:1122]: Is there really no way to do this via the accompanying rooms? :(
I assume the shirks hit you if you do this? 
grafik

And this seems a bit too much
grafik

Comment (JeffGainsNGames) [reply randovania/games/am2r/logic_database/Industrial Complex.txt:1122]: I tried both to see if movement was possible and it is not, the only thing that I can think of would be use a PB as soon as you enter the room.

File: randovania/games/am2r/logic_database/Industrial Complex.txt [hunk 3]
```diff
@@ -1108,7 +1108,24 @@ Extra - minimap_data: [{'x': 67, 'y': 29}, {'x': 67, 'y': 30}, {'x': 68, 'y': 28
...
+          Charged Bomb Jump (Intermediate) and Can Use Charged Bomb Jump
+          All of the following:
+              Can Use Spider Ball
+              Any of the following:
+                  Bombs or Spring Ball or Mid-Air Morph (Beginner)
+                  Power Bombs and Disabled Entrance Rando
+          # Damage Boost off of Shirk
+          Damage Boost (Expert) and Walljump (Advanced) and Normal Damage ≥ 50
+          # Shinespark
+          Speed Booster and Shinesparking Tricks (Beginner) and Short Charge (Expert)
```
Comment (Miepee) [inline randovania/games/am2r/logic_database/Industrial Complex.txt:1128]: just noticed, shouldnt this have a "short charge OR room rando"? afaik you can just charge from the right room  and kiill both the autoad and the shirk with the speedboost/shinespark

Comment (JeffGainsNGames) [reply randovania/games/am2r/logic_database/Industrial Complex.txt:1128]: there is not enough room, you will just run into the autoad unless we also want to add weapon requirements on top of `not entrance rando`

Comment (Miepee) [reply randovania/games/am2r/logic_database/Industrial Complex.txt:1128]: would rather not. but couldve sworn that it was possible...

**Review by Miepee — APPROVED** [review]

**Pull request merged**